--- src/GNUmakefile-orig
+++ src/GNUmakefile
@@ -41,7 +41,7 @@ ifdef SUMO
 	$(RANLIB) $(LIBDIR)/libnwchem-sumo.a
 endif
 	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL_MSG)
+#	$(INSTALL_MSG)
 #
 # This to link only and avoid tedious subdir makes
 #
@@ -62,7 +62,7 @@ ifdef SUMO
 	$(RANLIB) $(LIBDIR)/libnwchem-sumo.a
 endif
 #	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL_MSG)
+#	$(INSTALL_MSG)

 ccalink:
 	@echo "NWCHEM_CORE_LIBS = $(LIBS)" > ./cca/makefile.h
@@ -79,7 +79,7 @@ ifeq ($(TARGET),SOLARIS)
 endif
 	$(LINK.f) -p -o nwchem_prof nwchem.o stubs.o -lnwctask $(LIBS)
 	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL_MSG)
+#	$(INSTALL_MSG)
 #
 # This for linking test version
 #
@@ -88,7 +88,7 @@ ifeq ($(TARGET),SOLARIS)
 	/bin/rm -f nwchem_test
 endif
 	$(LINK.f) -o nwchem_test nwchem.o stubs.o -lnwctask $(LIBS)
-	$(INSTALL_MSG)
+#	$(INSTALL_MSG)

 configure_ga:
 ifdef OLD_GA
